src/org/gk/slicing/RevisionDetector.java

```diff
@@ -0,0 +1,442 @@
+
+		for (String attrName : revisionList) {
+			if (attributesRevised(reactionlikeEvent, getInstance(targetDBA, reactionlikeEvent), attrName))
+				return true;
+		}
+
+		// Check if a catalyst or regulator is added or removed.
+		Collection<String> additionsOrDeletionsList = Arrays.asList(ReactomeJavaConstants.regulatedBy,
+				ReactomeJavaConstants.catalystActivity);
+
+		for (String attrName : additionsOrDeletionsList) {
+			if (additionOrDeletionInList(reactionlikeEvent, getInstance(targetDBA, reactionlikeEvent), attrName))
+				return true;
+		}
+
+		// Check if summation is revised.
+		// Requires "special case" (examining "text" attribute for all summations).
+		if (isSummationRevised(reactionlikeEvent, targetDBA))
+			return true;
+
+		return false;
+	}
+
+	/**
+	 * Compare the value of a given attribute between two instances.
+	 *
+	 * Valid attribute checks take place in
+	 * {@link org.gk.slicing.SlicingEngine#isValidAttributeOrThrow(SchemaAttribute attribute)}
+	 *
+	 * @param oldInstance
+	 * @param newInstance
+	 * @param attrName
+	 * @return boolean (true if an attribute value is revised, false if all attribute values are equal).
+	 * @throws Exception
+	 */
+	private boolean attributesRevised(GKInstance oldInstance, GKInstance newInstance, String attrName)
+			throws Exception {
+		// Instance null checker.
+		if (oldInstance == null || newInstance == null) {
+			// if both instances are null, then there are no attributes to revise.
+			if (oldInstance == newInstance)
+				return false;
+			// if only one instance is null, then any attribute may be considered revised.
+			return true;
+		}
+
+		// Attribute null checker.
+		if (oldInstance.getAttributeValue(attrName) == null || newInstance.getAttributeValue(attrName) == null) {
+			if (oldInstance.getAttributeValue(attrName) == newInstance.getAttributeValue(attrName))
+				return false;
+			return true;
+		}
+
+		// If the object will not be able to cast to GKInstance, then simply compare the
+		// values.
+		if (oldInstance.getAttributeValue(attrName) instanceof String)
+			return !oldInstance.getAttributeValue(attrName).equals(newInstance.getAttributeValue(attrName));
+
+		// Default option is to cast to GKInstance and compare the database ID's.
+		Long oldInstanceDBID = ((GKInstance) oldInstance.getAttributeValue(attrName)).getDBID();
+		Long newInstanceDBID = ((GKInstance) newInstance.getAttributeValue(attrName)).getDBID();
+		return !oldInstanceDBID.equals(newInstanceDBID);
+	}
+
+	/**
+	 * Compare the attribute lists in order to detect additions or deletions.
+	 *
+	 * @param oldInstance
+	 * @param newInstance
+	 * @param attrName
+	 * @return boolean (true if there is an addition or deletion in the attribute list, false if the lists are equal).
+	 * @throws InvalidAttributeException
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	private boolean additionOrDeletionInList(GKInstance oldInstance, GKInstance newInstance, String attrName)
```

Comment:
What is the difference between this method and previous one, isChildPathwayAddedOrRemoved?